great_expectations/experimental/datasources/pandas_datasource.py

```diff
@@ -109,14 +111,14 @@ def get_batch_request(
         if options:
             for option, value in options.items():
                 if option in option_names_to_group and not isinstance(value, str):
-                    raise BatchRequestError(
+                    raise ge_exceptions.InvalidBatchRequestError(
                         f"All regex matching options must be strings. The value of '{option}' is "
                         f"not a string: {value}"
                     )
         return super().get_batch_request(options)
 
-    def _option_name_to_regex_group_id(self) -> Dict[str, int]:
-        option_to_group: Dict[str, int] = dict(self.regex.groupindex)
+    def _option_name_to_regex_group_id(self) -> BatchRequestOptions:
+        option_to_group: BatchRequestOptions = dict(self.regex.groupindex)
```

Comment:
Is this a useful change?
Does `BatchRequestOptions` have any structure that helps with type-narrowing?

Reply:
@Kilo59 In both `PathStr` and `BatchRequestOptions` changes that I introduced, I was striving for uniformity and consistency.  Having so many different type hints that mean very roughly the same things will make extending the code virtually impossible for anybody knew, and we will be constantly answering the question of "what belongs where".  Whenever we are dealing with `BatchRequest` and its components, there should be a single set of unambiguous types that developers will have an easy way of memorizing and using without constantly thinking about them or looking them up.  With that thinking, maybe we should consider removing the now confusing `PathStr` type alias and simply uses the `Union` explicitly where it is applicable.  Otherwise, I am seeing a growing number of abstract types and aliases that is making the code very difficult to follow.  Happy to discuss.  Thanks.

Reply:
Ok, I'd like to try to limit the scope of this discussion.

I'm okay with the `BatchRequestOptions` change.

The issue with `PathStr` is that it is being used incorrectly.
If an argument is only ever a `pathlib.Path` there's no reason to use a `pathlib.Path | str` union, it being a `TypeAlias` is a separate issue entirely.

Perhaps the `PathStr` would be better named as `PathOrStr` or as you said maybe we stop using type-aliases. But I think we should discuss this separately as a team.